engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -120,14 +126,36 @@ object MaterializeWorkflowDescriptorActor {
         }
       }
 
+      val errorOrMaybePrefixes = workflowOptions.getVectorOfStrings("call_cache_hit_path_prefixes")
+      val errorOrInvalidateBadCacheResults = errorOrCallCachingBoolean("invalidate-bad-cache-results")
+      val errorOrLogCacheHitFailures = errorOrCallCachingBoolean("log-cache-hit-failures")
+      val errorOrAddCacheHitFailuresToMetadata = errorOrCallCachingBoolean("add-cache-hit-failures-to-metadata")
+      val errorOrCallCachingOptions = (
```

Comment:
If we do stick to `ErrorOr`s, could we do the "config" part "once per instance" instead of "once per workflow"?

Reply:
That makes sense. My only hesitation is the changing logic. Currently the logic only crashes-on-a-bad-config for workflows that request call caching. I can take a shot at updating this and add a line to the ChangeLog and docs describing the change in behavior.